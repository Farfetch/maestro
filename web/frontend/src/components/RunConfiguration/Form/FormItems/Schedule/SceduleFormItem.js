import { Col, Form, Row, Select, Space, Switch, TimePicker } from "antd";
import { useEffect, useState } from "react";

const TitleFormItem = ({ initialScheduleEnabled }) => {
  const [isDaysTimeDisabled, setIsDaysTimeDisabled] = useState(null);

  useEffect(() => {
    setIsDaysTimeDisabled(!initialScheduleEnabled);
  }, [initialScheduleEnabled]);

  return (
    <>
      <Space style={{ marginBottom: "10px" }}>
        <Form.Item valuePropName="checked" name="isScheduleEnabled" noStyle>
          <Switch onChange={(value) => setIsDaysTimeDisabled(!value)} />
        </Form.Item>
        <span>Enable running schedule</span>
      </Space>

      <Row gutter={[12, 0]}>
        <Col span={12}>
          <Form.Item
            label="Days:"
            name="scheduleDays"
            rules={
              isDaysTimeDisabled
                ? []
                : [
                    {
                      required: true,
                      message: "Please select at least one day!"
                    }
                  ]
            }
          >
            <Select
              mode="multiple"
              allowClear
              style={{ width: "100%" }}
              placeholder="Select days"
              disabled={isDaysTimeDisabled}
            >
              <Select.Option key="Mon">Mon</Select.Option>
              <Select.Option key="Tue">Tue</Select.Option>
              <Select.Option key="Wed">Wed</Select.Option>
              <Select.Option key="Thu">Thu</Select.Option>
              <Select.Option key="Fri">Fri</Select.Option>
              <Select.Option key="Sat">Sat</Select.Option>
              <Select.Option key="Sun">Sun</Select.Option>
            </Select>
          </Form.Item>
        </Col>
        <Col span={12}>
          <Form.Item
            label="Time:"
            name="scheduleTime"
            rules={
              isDaysTimeDisabled
                ? []
                : [
                    {
                      required: true,
                      message: "Please select time to run!"
                    }
                  ]
            }
          >
            <TimePicker
              minuteStep={5}
              format="HH:mm"
              style={{ width: "100%" }}
              disabled={isDaysTimeDisabled}
            />
          </Form.Item>
        </Col>
      </Row>
    </>
  );
};

export default TitleFormItem;
