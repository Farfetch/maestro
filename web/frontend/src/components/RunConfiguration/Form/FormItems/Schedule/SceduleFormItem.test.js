import { cleanup, render, waitFor } from "@testing-library/react";
import { Form } from "antd";
import React from "react";

import SceduleFormItem from "./SceduleFormItem";

afterEach(cleanup);

describe("components/RunConfiguration/Form/SceduleFormItem", () => {
  test(`should render SceduleFormItem component`, async () => {
    const { container } = render(
      <Form>
        <SceduleFormItem />
      </Form>
    );

    await waitFor(() => {
      expect(container.outerHTML).toMatchSnapshot();
    });
  });
});
