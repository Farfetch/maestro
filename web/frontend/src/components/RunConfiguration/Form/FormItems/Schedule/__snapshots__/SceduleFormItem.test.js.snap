// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RunConfiguration/Form/SceduleFormItem should render SceduleFormItem component 1`] = `"<div><form class=\\"ant-form ant-form-horizontal css-dev-only-do-not-override-nnuwmp\\"><div class=\\"ant-space css-dev-only-do-not-override-nnuwmp ant-space-horizontal ant-space-align-center\\" style=\\"margin-bottom: 10px;\\"><div class=\\"ant-space-item\\" style=\\"margin-right: 8px;\\"><button id=\\"isScheduleEnabled\\" type=\\"button\\" role=\\"switch\\" aria-checked=\\"false\\" class=\\"ant-switch css-dev-only-do-not-override-nnuwmp\\"><div class=\\"ant-switch-handle\\"></div><span class=\\"ant-switch-inner\\"><span class=\\"ant-switch-inner-checked\\"></span><span class=\\"ant-switch-inner-unchecked\\"></span></span></button></div><div class=\\"ant-space-item\\"><span>Enable running schedule</span></div></div><div class=\\"ant-row css-dev-only-do-not-override-nnuwmp\\" style=\\"margin-left: -6px; margin-right: -6px;\\"><div style=\\"padding-left: 6px; padding-right: 6px;\\" class=\\"ant-col ant-col-12 css-dev-only-do-not-override-nnuwmp\\"><div class=\\"ant-form-item css-dev-only-do-not-override-nnuwmp\\"><div class=\\"ant-row ant-form-item-row css-dev-only-do-not-override-nnuwmp\\"><div class=\\"ant-col ant-form-item-label css-dev-only-do-not-override-nnuwmp\\"><label for=\\"scheduleDays\\" class=\\"\\" title=\\"Days:\\">Days</label></div><div class=\\"ant-col ant-form-item-control css-dev-only-do-not-override-nnuwmp\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><div class=\\"ant-select ant-select-in-form-item css-dev-only-do-not-override-nnuwmp ant-select-multiple ant-select-allow-clear ant-select-show-arrow ant-select-disabled ant-select-show-search\\" style=\\"width: 100%;\\"><div class=\\"ant-select-selector\\"><div class=\\"ant-select-selection-overflow\\"><div class=\\"ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix\\" style=\\"opacity: 1;\\"><div class=\\"ant-select-selection-search\\" style=\\"width: 0px;\\"><input type=\\"search\\" id=\\"scheduleDays\\" autocomplete=\\"off\\" class=\\"ant-select-selection-search-input\\" role=\\"combobox\\" aria-label=\\"Search\\" aria-expanded=\\"false\\" aria-haspopup=\\"listbox\\" aria-owns=\\"scheduleDays_list\\" aria-autocomplete=\\"list\\" aria-controls=\\"scheduleDays_list\\" readonly=\\"\\" unselectable=\\"on\\" style=\\"opacity: 0;\\" value=\\"\\" disabled=\\"\\"><span class=\\"ant-select-selection-search-mirror\\" aria-hidden=\\"true\\">&nbsp;</span></div></div></div><span class=\\"ant-select-selection-placeholder\\">Select days</span></div><span class=\\"ant-select-arrow\\" style=\\"user-select: none;\\" unselectable=\\"on\\" aria-hidden=\\"true\\"><span role=\\"img\\" aria-label=\\"down\\" class=\\"anticon anticon-down ant-select-suffix\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z\\"></path></svg></span></span></div></div></div></div></div></div></div><div style=\\"padding-left: 6px; padding-right: 6px;\\" class=\\"ant-col ant-col-12 css-dev-only-do-not-override-nnuwmp\\"><div class=\\"ant-form-item css-dev-only-do-not-override-nnuwmp\\"><div class=\\"ant-row ant-form-item-row css-dev-only-do-not-override-nnuwmp\\"><div class=\\"ant-col ant-form-item-label css-dev-only-do-not-override-nnuwmp\\"><label for=\\"scheduleTime\\" class=\\"\\" title=\\"Time:\\">Time</label></div><div class=\\"ant-col ant-form-item-control css-dev-only-do-not-override-nnuwmp\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><div class=\\"ant-picker css-dev-only-do-not-override-nnuwmp ant-picker-disabled\\" style=\\"width: 100%;\\"><div class=\\"ant-picker-input\\"><input id=\\"scheduleTime\\" readonly=\\"\\" placeholder=\\"Select time\\" title=\\"\\" size=\\"10\\" autocomplete=\\"off\\" value=\\"\\" disabled=\\"\\"><span class=\\"ant-picker-suffix\\"><span role=\\"img\\" aria-label=\\"clock-circle\\" class=\\"anticon anticon-clock-circle\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"clock-circle\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z\\"></path><path d=\\"M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z\\"></path></svg></span></span></div></div></div></div></div></div></div></div></div></form></div>"`;
